### PR TITLE
Expose getServerInfo API

### LINKED ISSUE #17438: Expose server information
To complete this https://github.com/microsoft/azuredatastudio/issues/20576, server information needs to be exposed from vscode-mssql.

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR fixes #17438 .
By exposing server information, target platform information can be set automatically for the sql database projects that are created from databases.

### CODE REVIEW

File: typings/vscode-mssql.d.ts [hunk 1]
```diff
@@ -114,6 +114,67 @@ declare module 'vscode-mssql' {
 		 * @returns A promise object for when the request receives a response
 		 */
 		sendRequest<P, R, E, R0>(requestType: RequestType<P, R, E, R0>, params?: P): Promise<R>;
+
+		/**
+		 * Get the server info for a connection
+		 * @param connectionInfo connection info of the connection
```
Comment (kisantia) [inline typings/vscode-mssql.d.ts:120]: also add `@returns` doc comment

**Review by kisantia — APPROVED** [review]

File: src/models/contracts/connection.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export namespace ConnectionCompleteNotification {
 /**
  * Information about a SQL Server instance.
  */
-export class ServerInfo {
+export class ServerInfo implements IServerInfo {
```
Comment (Charles-Gagnon) [inline src/models/contracts/connection.ts:45]: I'd just remove this class and replace it with a `ServerInfo` interface in the typings. There isn't any reason this needs to be a class anyways.

File: typings/vscode-mssql.d.ts [hunk 3]
```diff
@@ -114,6 +114,68 @@ declare module 'vscode-mssql' {
...
+		 * @returns server information
+	 	*/
+		getServerInfo(connectionInfo: IConnectionInfo): IServerInfo
+	}
+
+	/**
+	 * Information about a SQL Server instance.
+	 */
+	export interface IServerInfo {
```
Comment (alanrenmsft) [inline typings/vscode-mssql.d.ts:129]: we now have two places defining the ServerInfo, should we remove the old one and use this instead?

File: src/models/contracts/connection.ts (hunk 2 above)
Comment (SakshiS-harma) [reply src/models/contracts/connection.ts]: Makes sense. Removed it.

File: typings/vscode-mssql.d.ts (hunk 3 above)
Comment (SakshiS-harma) [reply typings/vscode-mssql.d.ts]: Yes, removed the old one.

**Review by Charles-Gagnon — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
(you need to fix the linting errors first though)

**Pull request merged**